src/pipecat/transports/services/daily.py

```diff
@@ -291,6 +289,7 @@ def __init__(
         self._transcription_ids = []
         self._transcription_status = None
 
+        self._joining = False
```

Comment:
Why do we need this? I had this before, but since we now guarantee that `join()` block until completed. There shouldn't be a case where you try to join more than once concurrently.